js/woo.js

```diff
@@ -0,0 +1,1586 @@
+        });
+        const response = await this[method] (this.extend (request, query));
+        //
+        // {
+        //     success: true,
+        //     rows: [
+        //         {
+        //             symbol: "SPOT_BTC_USDT",
+        //             side: "SELL",
+        //             executed_price: 46222.35,
+        //             executed_quantity: 0.0012,
+        //             executed_timestamp: "1641241162.329"
+        //         },
+        //         {
+        //             symbol: "SPOT_BTC_USDT",
+        //             side: "SELL",
+        //             executed_price: 46222.35,
+        //             executed_quantity: 0.0012,
+        //             executed_timestamp: "1641241162.329"
+        //         },
+        //         {
+        //             symbol: "SPOT_BTC_USDT",
+        //             side: "BUY",
+        //             executed_price: 46224.32,
+        //             executed_quantity: 0.00039,
+        //             executed_timestamp: "1641241162.287"
+        //         },
+        //         ...
+        //      ]
+        // }
+        //
+        const resultResponse = this.safeValue (response, 'rows', {});
+        return this.parseTrades (resultResponse, market, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        //
+        // ### public/market_trades
+        // {
+        //     symbol: "SPOT_BTC_USDT",
+        //     side: "SELL",
+        //     executed_price: 46222.35,
+        //     executed_quantity: 0.0012,
+        //     executed_timestamp: "1641241162.329"
+        // },
+        //
+        // ### fetchOrderTrades, fetchOrder->Transactions[]
+        //
+        // {
+        //     id: '99119876',
+        //     symbol: 'SPOT_WOO_USDT',
+        //     fee: '0.0024',
+        //     side: 'BUY',
+        //     executed_timestamp: '1641481113.084',
+        //     order_id: '87001234',
+        //     order_tag: 'default', <-- this param only in "fetchOrderTrades"
+        //     executed_price: '1',
+        //     executed_quantity: '12',
+        //     fee_asset: 'WOO',
+        //     is_maker: '1'
+        // }
+        //
+        const isFromFetchOrder = ('id' in trade);
+        const timestamp = this.safeTimestamp (trade, 'executed_timestamp');
+        const marketId = this.safeString (trade, 'symbol');
+        market = this.safeMarket (marketId, market, '_');
+        const symbol = market['symbol'];
+        const price = this.safeString (trade, 'executed_price');
+        const amount = this.safeString (trade, 'executed_quantity');
+        const order_id = this.safeString (trade, 'order_id');
+        const feeValue = this.safeString (trade, 'fee');
+        const feeAsset = this.safeString (trade, 'fee_asset');
+        const cost = Precise.stringMul (price, amount);
+        let side = this.safeString (trade, 'side');
+        if (side !== undefined) {
+            side = side.toLowerCase ();
+        }
```

Comment:
We can replace the above 3 lines with a one-liner `const side = this.safeStringLower (trade, 'side');`